rootfs/etc/profile.d/aws-vault.sh

```diff
@@ -35,16 +35,29 @@ if [ "${AWS_VAULT_ENABLED}" == "true" ]; then
 		fi
 	}
 
+	function choose_role_interactive() {
+		_preview="${FZF_PREVIEW:-crudini --format=ini --get "$AWS_CONFIG_FILE" 'profile {}'}"
+		crudini --get "${AWS_CONFIG_FILE}" |
+			awk -F ' ' '{print $2}' |
+			fzf \
+				--height 30% \
+				--reverse \
+				--prompt='-> ' \
+				--header 'Select AWS profile' \
+				--query "${AWS_ORG:-${NAMESPACE}}-" \
+				--preview "$_preview"
+	}
+
 	# Start a shell or run a command with an assumed role
 	function aws_vault_assume_role() {
-		role=${1:-${AWS_DEFAULT_PROFILE}}
-
 		# Do not allow nested roles
 		if [ -n "${AWS_VAULT}" ]; then
 			echo "Type '$(green exit)' before attempting to assume another role"
 			return 1
 		fi
 
+		role=${1:-$(choose_role_interactive)}
```

Comment:
Do we break back compatibility?

Reply:
@goruha Doesn't break backwards compat AFAIK, when running `assume-role` rather than defaulting to use `AWS_DEFAULT_PROFILE` , we need to select the profile to use. See attached gif showing usage.